lib/models/rules.js

```diff
@@ -132,6 +134,23 @@ function putR2core(rules, callback) {
     myutils.requestHelperWOMetrics('put', {url: config.perseoCore.rulesURL, json: rulesAndContexts}, callback);
 }
 
+/**
+ * Add ruleName if necessary in rule text field
+ *
+ * @param rule The rule object
+ */
+function normalizeRuleName(rule) {
+
+    var newAs = '"' + rule.name + '" as ruleName';
+    // Add "name as ruleName" if not exist
+    if (rule.text && rule.text.indexOf(' as ruleName') === -1) {
+        var idx = rule.text.toLowerCase().indexOf('select') + 7;
```

Comment:
7 seems to be a magic number, a bit obscure :)

Where it comes from? It seems to be len(select) +1, so maybe it would be more clear (i.e. less magical ;) something like this:

```
var offset = length('select') + 1;
var idx = rule.text.toLowerCase().indexOf('select') + offset;
```

Reply:
Fixed in b25bd0f